lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -1007,8 +981,61 @@ public void logCurrentState() {
         }
         logString.append("Finished logging current state. Time = ").append(currentTimeMillis());
         log.error("Current State: {}", logString.toString());
+
+        log.error(logString.toString());
+        logCurrentHeldLocks();
+        logOutstandingLockRequests();
     }
 
+    private void logAllHeldAndOutstandingLocks() {
+        LockServiceStateLogger lockServiceStateLogger = new LockServiceStateLogger(heldLocksTokenMap, outstandingLockRequestMultimap);
+        lockServiceStateLogger.logHeldLocks();
+        lockServiceStateLogger.logOutstandingLockRequests();
+    }
+
+    private void logOutstandingLockRequests() {
+        StringBuilder logString = new StringBuilder("Outstanding locks: \n");
+        for (Entry<LockClient, Collection<LockRequest>> lockClientToRequests : outstandingLockRequestMultimap.asMap().entrySet()) {
+            logString.append("Lock Client: ").append(lockClientToRequests.getKey().toString()).append("\n");
+            for (LockRequest request : lockClientToRequests.getValue()) {
+                logString.append("\t").append("Lock request: ").append(request.toString()).append("\n");
+                for (LockWithMode lock : request.getLocks()) {
+                    logString.append("\t\t").append("Lock descriptor: ").append(lock.toString()).append("\n");
```

Comment:
@schlosna if we don't log the descriptors, how will we know what lock is being held?  The entire point is to find the contended locks.  What particularly sensitive info are you worried about here?